test/unit/dbt_helper_test.py

```diff
@@ -204,65 +645,65 @@ def dbt_schema_mocks(mocker):
                 data_tests=None,
                 tables=[
                     DbtTable(
-                        name=str(mocker.sentinel.table_name),
-                        data_tests=[
-                            {
-                                "check_row_counts_per_partition": {
-                                    "table_name": str(mocker.sentinel.table_name),
-                                    "partition_column": str(
-                                        mocker.sentinel.partition_column
-                                    ),
-                                }
-                            }
-                        ],
+                        name=table_name,
+                        data_tests=data_tests,
                         columns=[
                             DbtColumn(
-                                name=str(mocker.sentinel.field_name),
+                                name=field_name,
                             ),
                         ],
                     ),
                 ],
             ),
         ],
     )
-    yaml = f"""
+
+    yaml_tests = (
+        f"""\
+        data_tests:
+          - check_row_counts_per_partition:
+              table_name: {table_name}
+              partition_column: {partition_column}
+        """
+        if has_row_count_test
+        else ""
+    )
+
+    yaml = f"""\
 version: 2
 sources:
   - name: pudl
     tables:
-      - name: {mocker.sentinel.table_name}
-        data_tests:
-          - check_row_counts_per_partition:
-              table_name: {mocker.sentinel.table_name}
-              partition_column: {mocker.sentinel.partition_column}
+      - name: {table_name}
+{yaml_tests.rstrip()}
         columns:
-          - name: {mocker.sentinel.field_name}
+          - name: {field_name}
 """
     return DbtSchemaMocks(
         field=mocked_field,
         resource=mocked_resource,
         pudl_package=mocked_ppkg,
-        table_name=str(mocker.sentinel.table_name),
-        partition_column=str(mocker.sentinel.partition_column),
+        table_name=table_name,
+        partition_column=partition_column,
         schema=schema,
         yaml=yaml,
+        has_row_count_test=has_row_count_test,
+        load_method=load_method,
     )
 
 
-def test_dbt_schema__from_table_name(dbt_schema_mocks):
-    actual = DbtSchema.from_table_name(
-        dbt_schema_mocks.table_name,
-        partition_column=dbt_schema_mocks.partition_column,
-    )
-    assert actual == dbt_schema_mocks.schema
-
-
-def test_dbt_schema__from_yaml(mocker, dbt_schema_mocks):
-    with StringIO(dbt_schema_mocks.yaml) as f:
+def test_dbt_schema_loading(dbt_schema_mocks, mocker):
```

Comment:
clarification: this test checks to make sure we can round-trip the schemas through yaml and also through our bespoke metadata dictionary?

Reply:
Yes, exactly